packages/@uppy/xhr-upload/src/index.js

```diff
@@ -230,32 +230,23 @@ module.exports = class XHRUpload extends Plugin {
           const body = opts.getResponseData(xhr.responseText, xhr)
           const uploadURL = body[opts.responseUrlFieldName]
 
-          const response = {
+          const uploadResp = {
             status: ev.target.status,
             body,
             uploadURL
           }
 
-          this.uppy.setFileState(file.id, { response })
-
-          this.uppy.emit('upload-success', file, body, uploadURL)
+          this.uppy.emit('upload-success', file, uploadResp)
 
           if (uploadURL) {
             this.uppy.log(`Download ${file.name} from ${file.uploadURL}`)
           }
 
           return resolve(file)
         } else {
-          const body = opts.getResponseData(xhr.responseText, xhr)
+          // const body = opts.getResponseData(xhr.responseText, xhr)
```

Comment:
the response data was added intentionally in #612, i think we could keep the `setFileState` stuff here as a special case, at least for now…

Reply:
Why can’t we set response in Core, the same way we set it with `upload-success`?

Reply:
my thought was, that the upload-error event does not have a `response` parameter so we couldn't do it in core. but on second thought it would be fine to add that parameter.

Reply:
Yeah, I did this:

```js
this.on('upload-error', (file, error, response) => {
  this.setFileState(file.id, {
    error: error.message,
    response
  })
})

...

const body = opts.getResponseData(xhr.responseText, xhr)
const error = buildResponseError(xhr, opts.getResponseError(xhr.responseText, xhr))

const response = {
  status: ev.target.status,
  body
}

this.uppy.emit('upload-error', file, error, response)
return reject(error)
```

Reply:
But now I’m wondering, why is response in `upload-error` only returned once, in XHR’s `load` event and nowhere else. Because that’s the only place where something is returned from the server? What about S3 and tus?